src/lib/CLValue/KeyVariants.ts

```diff
@@ -658,10 +567,91 @@ export class BidAddr implements CLKeyVariant {
       default:
         throw new Error('Unsupported tag while deserializing!');
     }
+  },
+  toBytes(value: BidAddr): Uint8Array {
+    const tag = value.tag();
+
+    switch (tag) {
+      case BidAddrTag.Unified:
+        return concat([[tag], value.data.Unified!.data]);
+      case BidAddrTag.Validator:
+        return concat([[tag], value.data.Validator!.data]);
+      case BidAddrTag.Delegator:
+        return concat([
+          [tag],
+          value.data.Delegator!.validator.data,
+          value.data.Delegator!.delegator.data
+        ]);
+      case BidAddrTag.Credit:
+        const era = BigNumber.from(value.data.Credit!.eraId);
+        return concat([
+          [tag],
+          value.data.Credit!.validator.data,
+          toBytesU64(era)
+        ]);
+      default:
+        throw new Error('Unsupported tag while serializing!');
+    }
+  }
+};
+
+export class BidAddr implements CLKeyVariant {
+  keyVariant = KeyTag.BidAddr;
+  prefix = BID_ADDR_PREFIX;
+
+  bidAddrTag = BidAddrTag;
+
+  constructor(public data: BidAddrData) {}
+
+  value(): any {
```

Comment:
```suggestion
  value(): BidAddrData {
```

can we have type here?